File: app/src/main/java/ru/melod1n/vk/mvp/presenter/FriendsPresenter.java
package ru.melod1n.vk.mvp.presenter;

import androidx.annotation.NonNull;

import java.util.ArrayList;

import ru.melod1n.library.mvp.base.MvpFields;
import ru.melod1n.library.mvp.base.Presenter;
import ru.melod1n.vk.api.model.VKUser;
import ru.melod1n.vk.mvp.repository.FriendsRepository;
import ru.melod1n.vk.mvp.view.FriendsView;

public class FriendsPresenter extends Presenter<VKUser, FriendsView> {

    public static final String ONLY_ONLINE = "_only_online";

    public FriendsPresenter(@NonNull FriendsView view) {
        super(view);

        initRepository(new FriendsRepository());
    }


    @Override
    protected void insertValues(@NonNull MvpFields fields, @NonNull ArrayList<VKUser> values) {
//        if (view != null) {
            view.insertValues(fields, values);
//        }
    }


}

File: app/src/main/java/ru/melod1n/vk/mvp/presenter/FriendsPresenter.kt
package ru.melod1n.vk.mvp.presenter

import ru.melod1n.vk.api.model.VKUser
import ru.melod1n.vk.mvp.contract.BaseContract
import ru.melod1n.vk.mvp.contract.FriendsContract
import ru.melod1n.vk.mvp.repository.FriendsRepository

class FriendsPresenter(view: BaseContract.View<VKUser>) : FriendsContract.Presenter<VKUser>(view) {

    override val tag = "FriendsPresenter"

    override val repository = FriendsRepository()
    override var loadedValues = ArrayList<VKUser>()
    override var cachedValues = ArrayList<VKUser>()
}